enhancements/20250522-expose-cdi-settings-in-ui.md

```diff
@@ -0,0 +1,126 @@
+- `cdi.harvesterhci.io/storageProfileCloneStrategy`: String value, one of: "copy", "snapshot", "csi-clone"
+- `cdi.harvesterhci.io/storageProfileVolumeSnapshotClass`: String value, must be an existing VolumeSnapshotClass name
+- `cdi.harvesterhci.io/storageProfileVolumeModeAccessModes`: JSON string mapping VolumeMode to AccessMode arrays
+
+Below is the StorageClass yaml example:
+
+```yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  annotations:
+    cdi.harvesterhci.io/filesystemOverhead: "0.85"
+    cdi.harvesterhci.io/storageProfileCloneStrategy: "snapshot"
+    cdi.harvesterhci.io/storageProfileVolumeSnapshotClass: "csi-snapshot-class"
+    cdi.harvesterhci.io/storageProfileVolumeModeAccessModes: |
+      {"Block":["ReadWriteOnce","ReadOnlyMany"],"Filesystem":["ReadWriteOnce","ReadWriteMany"]}
+```
+
+The implementation consists of three main components:
+
+1. StorageClass Webhook Validation
+Extend the existing StorageClass validation webhook to validate the five new annotations. The webhook will perform pattern matching for filesystemOverhead, enum validation for cloneStrategy, existence checks for volumeSnapshotClass, and JSON schema validation for volumeModeAccessModes.
+2. New StorageClass Controller
+Introduce a new storageclass_controller specifically for watching for StorageClass changes and handling filesystemOverhead synchronization to the CDI object and synchronize cloneStrategy, volumeSnapshotClass, and volumeModeAccessModes annotations to the corresponding StorageProfile CRD.
+
+### Compatibility
+Currently, the storage_profile_controller automatically populates default values such as cloneStrategy, accessMode, and volumeMode for validated CSI drivers like LonghornV2 and LVM.
+
+For LonghornV2, the previous behavior sets the following defaults when a StorageProfile is created:
+- clone strategty is set to `copy`
+- volume snapshot class is set to `longhorn-snapshot`
+
+For LVM, the default values are:
+- volume mode is set to `Block`
+- access modes is set to `["ReadWriteOnce"]`
+
+This HEP preserves the same logic, but shifts it to the StorageClass mutator. The mutator will:
+- add annotations `cdi.harvesterhci.io/storageProfileCloneStrategy: "snapshot"` and `cdi.harvesterhci.io/storageProfileVolumeSnapshotClass: "longhorn-snapshot"` to LonghornV2 StorageClasses if both annotations are not set.
+- add `cdi.harvesterhci.io/storageProfileVolumeModeAccessModes: "{\"Block\":[\"ReadWriteOnce\"]}"` to LVM StorageClasses if the annotation is not already present.
+
+Note that Longhorn v2 will support CSI clone starting from v1.9.1, we need to set the default CDI clone strategy to `"clone"` in the storage class annotation `cdi.harvesterhci.io/storageProfileCloneStrategy`.
+
+### Test plan
+
+1. Create StorageClass with valid annotations and verify synchronization
+2. Update StorageClass annotations and verify changes propagate
+3. Delete StorageClass and verify cleanup behavior
```

Comment:
Does cleanup involve deleting the storage profile even though it is created by CDI?